sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,97 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient. They can be expressed as
+    a product of binomial coefficients or of factorials:
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2, 1, 0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3).rewrite(gamma)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        from sympy import binomial
+        if any(x.is_negative for x in args):
+            if len(args) == 2:
+                return(binomial(Add(*args), args[0]))
```

Comment:
`multinomial(a,b) == multinomial(b,a)` but `binomial(a+b,a) != binomial(a+b,b)` so I don't think you should have this special case:
```
>>> a,b=3,-4
>>> binomial(a+b,b)
0
>>> binomial(a+b,a)
-1
```
I wonder if the logic should be
```python
if any(x.is_negative and x.is_integer for x in k):
    return S.Zero
if all(x.is_Integer for x in k):
    return binomial product
if all(x.is_number for x in k):
    # check that indeterminant values don't sneak through
    if not any(x.is_integer is None or x.is_negative is None for x in k):
        return gamma form
remain unevaluated
```

Reply:
in the above logic `multinomial(1,-5)` gets ignored ... doesn't it ... but in actuality [it equals -4](http://www.wolframalpha.com/input/?i=multinomial(1%2C-5))

Reply:
If you want to link to the binomial result then you will have to be sure to use the positive argument when there are two:
```python
    if len(args) == 2:
                return(binomial(Add(*args), max(args)))
```